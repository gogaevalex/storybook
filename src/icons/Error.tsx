import { Svg, SvgProps } from '../components/Svg';

export const ErrorIcon = (props: SvgProps) => (
  <Svg {...props}>
    <path d="M11.25 16.25H13.75V18.75H11.25V16.25ZM11.25 6.25H13.75V13.75H11.25V6.25ZM12.4875 0C5.5875 0 0 5.6 0 12.5C0 19.4 5.5875 25 12.4875 25C19.4 25 25 19.4 25 12.5C25 5.6 19.4 0 12.4875 0ZM12.5 22.5C6.975 22.5 2.5 18.025 2.5 12.5C2.5 6.975 6.975 2.5 12.5 2.5C18.025 2.5 22.5 6.975 22.5 12.5C22.5 18.025 18.025 22.5 12.5 22.5Z" />
  </Svg>
);
