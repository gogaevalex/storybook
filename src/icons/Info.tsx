import { Svg, SvgProps } from '../components/Svg';

export const InfoIcon = (props: SvgProps) => (
  <Svg {...props}>
    <path d="M13.75 8.75L11.25 8.75L11.25 6.25L13.75 6.25L13.75 8.75ZM13.75 18.75L11.25 18.75L11.25 11.25L13.75 11.25L13.75 18.75ZM12.5125 25C19.4125 25 25 19.4 25 12.5C25 5.6 19.4125 4.88475e-07 12.5125 1.09169e-06C5.6 1.696e-06 -1.696e-06 5.6 -1.09278e-06 12.5C-4.89568e-07 19.4 5.6 25 12.5125 25ZM12.5 2.5C18.025 2.5 22.5 6.975 22.5 12.5C22.5 18.025 18.025 22.5 12.5 22.5C6.975 22.5 2.5 18.025 2.5 12.5C2.5 6.975 6.975 2.5 12.5 2.5Z" />
  </Svg>
);
