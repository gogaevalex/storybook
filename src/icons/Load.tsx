import { Svg, SvgProps } from '../components/Svg';

export const LoadIcon = (props: SvgProps) => (
  <Svg {...props}>
    <path d="M13.2964 21.2406C8.15069 21.2406 3.98892 17.3183 3.98892 12.4687C3.98892 8.04511 7.45928 4.41103 11.9668 3.79699V0C5.23878 0.626566 0 5.96491 0 12.4687C0 19.386 5.94349 25 13.2831 25C17.6842 25 21.5801 22.9825 24 19.8747L20.5429 17.9574C18.841 19.9624 16.2349 21.2406 13.2964 21.2406Z" />
  </Svg>
);
