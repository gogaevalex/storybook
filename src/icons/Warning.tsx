import { Svg, SvgProps } from '../components/Svg';

export const WarningIcon = (props: SvgProps) => (
  <Svg {...props}>
    <path d="M12.5 5.15625L21.0568 20.1349H3.94318L12.5 5.15625ZM12.5 0.5625L0 22.4375H25L12.5 0.5625ZM13.6364 16.6809H11.3636V18.9836H13.6364V16.6809ZM13.6364 9.77303H11.3636V14.3783H13.6364V9.77303Z" />
  </Svg>
);
