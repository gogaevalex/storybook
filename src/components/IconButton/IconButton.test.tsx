import { render } from '@testing-library/react';
import { TestProvider } from '../../test/TestProvider';
import { IconButton } from '.';
import { SuccessIcon } from '../../icons/Success';

describe('IconButton', () => {
  it('Default render', () => {
    const { container } = render(
      <TestProvider>
        <IconButton>
          <SuccessIcon />
        </IconButton>
      </TestProvider>
    );
    expect(container).toMatchSnapshot();
  });
});
