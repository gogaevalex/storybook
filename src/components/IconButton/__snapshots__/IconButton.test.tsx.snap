// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton Default render 1`] = `
<div>
  <button
    class="css-1phao6i"
    type="button"
  >
    <svg
      class="css-d1qd63"
      viewBox="0 0 25 25"
    >
      <path
        d="M12.5 0C5.6 0 0 5.6 0 12.5C0 19.4 5.6 25 12.5 25C19.4 25 25 19.4 25 12.5C25 5.6 19.4 0 12.5 0ZM12.5 22.5C6.9875 22.5 2.5 18.0125 2.5 12.5C2.5 6.9875 6.9875 2.5 12.5 2.5C18.0125 2.5 22.5 6.9875 22.5 12.5C22.5 18.0125 18.0125 22.5 12.5 22.5ZM18.2375 6.975L10 15.2125L6.7625 11.9875L5 13.75L10 18.75L20 8.75L18.2375 6.975Z"
      />
    </svg>
  </button>
</div>
`;
